set TableView dimensions from QML

vandenman opened the pull request.

vandenman committed:
enable setting table dimensions from QML

vandenman committed:
custom validator function & use itemTypePerItem to set default values

JorisGoosen commented on QMLComponents/models/listmodeltableviewbase.cpp:
```diff
@@ -180,6 +180,66 @@ void ListModelTableViewBase::removeRow(size_t row, bool emitStuff)
+	{
+		endResetModel();
+
+		emit columnCountChanged();
```
maybe only emit it when changed?

vandenman replied:
done

vandenman committed:
track if rows or columns change and only emit if they do

shun2wang commented:
is this related to [tableview](https://github.com/jasp-stats/jasp-desktop/blob/stable/Docs/development/jasp-qml-guide.md#tableview)?

@vandenman Could you please also add these componets or arguments into QML [developer guide](https://github.com/jasp-stats/jasp-desktop/blob/stable/Docs/development/jasp-qml-guide.md)?

Looks like this is what I was looking for, thanks.

JorisGoosen commented:
@vandenman how did you test this?
Something in the testmodule perhaps?

vandenman committed:
add some documentation

vandenman commented:
I tested it with the skf module, but that isn't online anywhere. I can push that, or should I copy-ish the tableview I used to the testmodule?

vandenman commented:
@shun2wang I just added some documentation, but the source code is unfortunately the best documentation we have right now. There are a bunch of other things possible with table views that I didn't go into, I only discussed the things I changed in this PR.

vandenman committed:
docs: itemTypePerColumn -> itemTypePerRow
